samples/client/petstore/scala-sttp/src/main/scala/org/openapitools/client/api/PetApi.scala

```diff
@@ -32,13 +33,15 @@ class PetApi(baseUrl: String) {
    * 
    * @param pet Pet object that needs to be added to the store
    */
-  def addPet(pet: Pet
-): Request[Either[ResponseError[Exception], Pet], Nothing] =
+  def addPet(pet: Pet): Request[Either[ResponseException[ApiModel, Exception], Pet], Any] =
```

Comment:
I think for `Left` case we should have more concrete generated type such as `ExceptionApiModel`
```scala
sealed trait ApiModel
case class Pet extends ApiModel
case class PetNotFound extends ExceptionApiModel
```

```suggestion
  def addPet(pet: Pet): Request[Either[ResponseException[ExceptionApiModel, Exception], Pet], Any] =
```

Reply:
I'm not sure if the change would be valid. The same model can be used for handling both 200 and 400 responses in theory.

Reply:
if `ApiModel` and `ExceptionApiModel` are traits (by default corresponding to 2xx and 4xx/5xx responses), then if a concrete model class is re-used, it could implemet both of these traits?